BSPConversionLib/Source/EntityConverter.cs

```diff
@@ -853,6 +855,14 @@ private void ConvertAngles(Entity entity)
 			}
 		}
 
+		private void SetTeleportOrigin(Entity teleDest)
+		{
+			string[] origin = teleDest["origin"].Split(' ');
+			float.TryParse(origin[2], out var z);
```

Comment:
This should be `float.Parse` instead of you are expecting the value to always parse successfully.